.github/workflows/go.yml

```diff
@@ -18,20 +18,25 @@ jobs:
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v5
+
       - name: Setup Go
         uses: actions/setup-go@v6
         with:
-          go-version: '1.24'
+          go-version-file: go.mod
+
       - name: Set up gotestfmt
         uses: GoTestTools/gotestfmt-action@v2
         with:
           repo: gotestfmt
           version: v2.5.0
+
       - name: Build
         run: make build
+
       - name: Unit Tests
         env:
           RUN_MEMORY_LEAK_TEST: true
         run: make test
-      - uses: golangci/golangci-lint-action@v8
```

Comment:
Why this change, to me it looks better when using official runner, they are responsible for downloading and making it available in the `$PATH`, this to me looks like complicated way of installing it with `Makefile` instead of relying on official way

Reply:
it makes sense if it's something we want to be able to run locally 

go doesn't have dependencies groups ? so we can put this tool within test group or something like that ?